stripe/functions/index.js

```diff
@@ -144,11 +144,17 @@ exports.cleanupUser = functions.auth.user().onDelete(async (user) => {
   const customer = (await dbRef.doc(user.uid).get()).data();
   await stripe.customers.del(customer.customer_id);
   // Delete the customers payments & payment methods in firestore.
-  const snapshot = await dbRef
+  const paymetsMethodsSnapshot = await dbRef
     .doc(user.uid)
     .collection('payment_methods')
     .get();
-  snapshot.forEach((snap) => snap.ref.delete());
+  paymetsMethodsSnapshot.forEach((snap) => snap.ref.delete());
```

Comment:
I'm realizing that we mistakenly weren't `await`-ing the delete here. That means that this function won't reliably delete all information. I believe we'll need to do something like:

```js
const deletes = [];

paymetsMethodsSnapshot.forEach((snap) => deletes.push(snap.ref.delete()));

// more code ...

paymentsSnapshot.forEach((snap) => deletes.push(snap.ref.delete()));

// the important part
await Promise.all(deletes);
```

Reply:
Yes, I totally agree with you. I missed that too.. thank you!
I think using an array is more simple than using `batch` as instructed in docs https://firebase.google.com/docs/firestore/manage-data/delete-data
how do you think about using batch though like below.
Is there any reason you used just array? 
```
const batch = adminfirestore().batch();
  const paymetsMethodsSnapshot = await dbRef
    .doc(user.uid)
    .collection('payment_methods')
    .get();
  paymetsMethodsSnapshot.forEach((snap) => batch.delete(snap.ref));
  const paymentsSnapshot = await dbRef
    .doc(user.uid)
    .collection('payments')
    .get();
  paymentsSnapshot.forEach((snap) => batch.delete(snap.ref));

  await batch.commit();
```

Reply:
Happy New Year @HYJ29! Great point, I think a `batch` delete definitely makes sense so that we can be confident all data was wiped out at once.

Reply:
Ok. I added it as said above! Happy new year! @jhuleatt